### PR TITLE
Develop r package

### PR DESCRIPTION
**Pull request by afwillia**
The goal of this PR is to create an R package framework for DCA. Allowing for easier testing and documentation using devtoosl, testthat, roxygen, etc. It adds R package skeleton documents to DCA to make it the `datacurator` package. It also adds schematic and synapse API functions to `R` with corresponding tests in `tests`.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Package: datacurator
+Version: 0.1.0
+Title: Sage Bionetwork's Metadata Ingress Shiny App
+Description: This Shiny app allows users to validate metadata files and upload to Synapse.
+Authors: Rongrong Chai, Xengie Doan, Milen Nikolov, Sujay Patil, Robert Allaway, Bruno Grande, Anthony Williams, Lauren Wolfe
```
Comment (lakikowolfe) [inline DESCRIPTION:5]: Small change here but can you please update my name: Lauren -> Loren

Comment (afwillia) [reply DESCRIPTION]: yes of course

**Comment by afwillia** [comment]
@rrchai

To run the schematic REST API unit tests, you'll need to set up the schematic server.

First, [install schematicpy according to their instructions](https://github.com/Sage-Bionetworks/schematic/#installation).

Clone the schematic repo and `cd` into it. Then run the schematic API in a terminal with `poetry run python run_api.py`. You should see the following lines

``` 
* Serving Flask app "api" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
```

Confirm you can access the API by navigating to http://0.0.0.0:3001/v1/ui/ in your browser. You should see a UI listing available endpoints.

When the server is running in a terminal, open Rstudio and set the working directory to the data_curator dir. Then run `devtools::test()`

**Comment by afwillia** [comment]
Closing this pull request because shinyapps.io loads shinyapps from R packages differently than the current format. I need to plan this refactor to an R package more carefully before proceeding.

**Pull request closed without merging**